skt/kernelbuilder.py

```diff
@@ -234,11 +249,80 @@ def assemble_make_options(self):
         """Assemble all of the make options into a list."""
         kernel_build_argv = (
             self.make_argv_base
-            + self.targz_pkg_argv
+            + self.compile_args
             + self.extra_make_args
         )
         return kernel_build_argv
 
+    def find_rpm(self):
+        """
+        Find RPMs in the buildlog.
+
+        Returns:
+            A list of paths to RPM files.
+
+        """
+        # Compile a regex to look for the tarball filename.
+        rpm_regex = re.compile(r"^Wrote: (.*\.rpm)$")
+
+        # Read the buildlog line by line looking for the RPMs.
+        rpm_files = []
+        with open(self.buildlog, 'r') as fileh:
+            for log_line in fileh:
+                match = rpm_regex.search(log_line)
+
+                # If we find a match, store the path to the tarball and stop
+                # reading the buildlog.
+                if match:
+                    rpm_path = match.group(1)
+                    rpm_files.append(rpm_path)
+
+                    # Ensure the RPM is actually on the filesystem.
+                    if not os.path.isfile(rpm_path):
+                        raise IOError(
+                            "Built kernel RPM not found: "
+                            "{}".format(rpm_path)
+                        )
+
+        return rpm_files
+
+    def make_rpm_repo(self, rpm_files):
+        """
+        Make a RPM repository within a directory.
+
+        Args:
+            rpm_files:  A list of RPM paths.
+
+        Returns:
+            Path to the repo directory.
+
+        """
+        # Set a path for the RPM repository directory.
+        repo_dir = join_with_slash(self.source_dir, 'rpm_repo/')
+
+        # Remove the existing repo directory if it exists.
+        if os.path.isdir(repo_dir):
+            shutil.rmtree(repo_dir)
+
+        # Create the directory.
+        os.mkdir(repo_dir)
+
+        # Move our current RPMs to that directory.
+        for rpm_file in rpm_files:
+            shutil.move(rpm_file, repo_dir)
+
+        # Create an RPM repository in the repo directory.
+        args = ["createrepo", repo_dir]
+        exit_code = self.run_multipipe(args)
+
+        if exit_code != 0:
+            raise(
+                IOError,
+                "Unable to create RPM repository at {}".format(repo_dir)
+            )
```

Comment:
Maybe
```python
            raise subprocess.CalledProcessError(
                exit_code,
                ' '.join(args)
            )
```
Should be better in this case

Reply:
That does make more sense. I'll change it.